### PR TITLE
fix(cypress): notifications editor flaky tests

### PR DESCRIPTION
**Pull request by p-fernandez**
- **What kind of change does this PR introduce?** (Bug fix)

Some fixes for the tests in Cypress of Notifications editor. It seems to happen some race conditions where in the tests access to elements was tried before them fully load. Pattern quite repeatedly in less powerful dev environments.

- **Why was this change needed?** (You can also link to an open issue here)

Improve the efficiency of the CI. Though at the cost of a bit of speed in test executions.

- **Other information**:
N/A

### CODE REVIEW

File: apps/web/cypress/tests/notifications-editor.spec.ts [hunk 1]
```diff
@@ -213,6 +216,8 @@ describe('Notifications Creator', function () {
       waitLoadTemplatePage(() => {
         cy.visit('/templates/create');
       });
+      cy.waitForNetworkIdle(500);
```
Comment (p-fernandez) [inline apps/web/cypress/tests/notifications-editor.spec.ts:219]: In slow machines sometimes the page wasn't loaded properly.

File: apps/web/cypress/tests/notifications-editor.spec.ts [hunk 2]
```diff
@@ -352,7 +360,6 @@ describe('Notifications Creator', function () {
       cy.getByTestId('feed-button-1-checked');
       cy.getByTestId('create-feed-input').type('test4');
       cy.getByTestId('add-feed-button').click();
-      cy.wait(1000);
```
Comment (p-fernandez) [inline apps/web/cypress/tests/notifications-editor.spec.ts:355]: Found this is not needed.

File: apps/web/cypress/tests/notifications-editor.spec.ts [hunk 3]
```diff
@@ -554,7 +561,7 @@ describe('Notifications Creator', function () {
       cy.get('#codeEditor').contains('Hello world code {{name}} <div>Test</div>');
     });
 
-    it('should redirect to dev env for edit template', async function () {
```
Comment (p-fernandez) [inline apps/web/cypress/tests/notifications-editor.spec.ts:557]: Cypress reported a warning/error from defining the test function async.

File: apps/web/cypress/tests/notifications-editor.spec.ts [hunk 4]
```diff
@@ -568,10 +575,10 @@ describe('Notifications Creator', function () {
         cy.intercept('GET', '/v1/changes?promoted=false').as('unpromoted-changes');
         cy.visit('/changes');
         cy.wait('@unpromoted-changes');
-        cy.getByTestId('promote-btn').eq(0).click();
 
+        // TODO: Race condition around here, makes test flaky.
```
Comment (p-fernandez) [inline apps/web/cypress/tests/notifications-editor.spec.ts:579]: This one I wasn't able to find out how to make it not flaky.

**Review by scopsy — APPROVED** [review]
Crossing my fingers it will fix the issues there 🤞 Thanks for picking this up

**Comment by scopsy** [comment]
Ofcourse it failed 😅

**Comment by p-fernandez** [comment]
> Ofcourse it failed 😅

The one I was suspecting. 🤦🏻 
If someone knows a bit more that feature and could explain me what it does I think I could guess what's the potential race condition. With the other tests it was less hard to notice it.

**Comment by scopsy** [comment]
@p-fernandez when creating a notification template, you promote the changes of this template to production. 
Then switch to the production environment and click on the template edit button.

Once you are in the template editor screen on the `Production` environment, switch the environment flag to be `Development` This should automatically redirect the user to the `Development` template and basically have a different URL. This is what the test is testing.

**Comment by p-fernandez** [comment]
I think I identified where the flakiness comes from.
The environment selector has a behaviour that before being available has an overlay on top and the size of the input radios (the element that renders every environment) has a size of 0x0 so Cypress can't pick them up to run a click there. I couldn't debug it very deep but it also looks like the environment selector re-renders multiple times when loading and has some API calls associated that might make it unavailable for Cypress tests.
I need more investigation, but it is the line I think I will follow. Hope it is not a red herring.

**Comment by scopsy** [comment]
@p-fernandez might be it. Is it possible to wait until the element is visible and the overlay is not visible? Maybe with a combination of waitForIdle or something similar. Let me know if you will need help to go over it together in sync 🚀

**Comment by p-fernandez** [comment]
@scopsy I gave many runs in a row and seems only to fail when uploading artifacts. Tests seem quite stable (hope I am not jinxing it).

**Review by scopsy — APPROVED** [review]
Let's give a try in the real world ;P Great work here!

**Pull request merged**